WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/ProductPriceSettingsViewController.swift

```diff
@@ -0,0 +1,416 @@
+//
+private extension ProductPriceSettingsViewController {
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as UnitInputTableViewCell where row == .price:
+            configurePrice(cell: cell)
+        case let cell as UnitInputTableViewCell where row == .salePrice:
+            configureSalePrice(cell: cell)
+        case let cell as SwitchTableViewCell where row == .scheduleSale:
+            configureScheduleSale(cell: cell)
+        case let cell as SettingTitleAndValueTableViewCell where row == .scheduleSaleFrom:
+            configureScheduleSaleFrom(cell: cell)
+        case let cell as DatePickerTableViewCell where row == .datePickerSaleFrom:
+            configureSaleFromPicker(cell: cell)
+        case let cell as SettingTitleAndValueTableViewCell where row == .scheduleSaleTo:
+            configureScheduleSaleTo(cell: cell)
+        case let cell as DatePickerTableViewCell where row == .datePickerSaleTo:
+            configureSaleToPicker(cell: cell)
+        case let cell as SettingTitleAndValueTableViewCell where row == .taxStatus:
+            configureTaxStatus(cell: cell)
+        case let cell as SettingTitleAndValueTableViewCell where row == .taxClass:
+            configureTaxClass(cell: cell)
+        default:
+            fatalError()
+            break
+        }
+    }
+
+    func configurePrice(cell: UnitInputTableViewCell) {
+        let viewModel = Product.createRegularPriceViewModel(regularPrice: regularPrice, using: CurrencySettings.shared) { [weak self] value in
+            self?.handleRegularPriceChange(regularPrice: value)
+        }
+        cell.selectionStyle = .none
+        cell.configure(viewModel: viewModel)
+    }
+
+    func configureSalePrice(cell: UnitInputTableViewCell) {
+        let viewModel = Product.createSalePriceViewModel(salePrice: salePrice, using: CurrencySettings.shared) { [weak self] value in
+            self?.handleSalePriceChange(salePrice: value)
+        }
+        cell.selectionStyle = .none
+        cell.configure(viewModel: viewModel)
+    }
+
+    func configureScheduleSale(cell: SwitchTableViewCell) {
+        cell.selectionStyle = .none
+        cell.title = NSLocalizedString("Schedule sale", comment: "Title of the cell in Product Price Settings > Schedule sale")
+        cell.subtitle = NSLocalizedString("Automatically start and end a sale", comment: "Subtitle of the cell in Product Price Settings > Schedule sale")
+        cell.isOn = (dateOnSaleStart != nil && dateOnSaleEnd != nil) ? true : false
+        cell.onChange = { [weak self] isOn in
+            guard let self = self else {
+                return
+            }
+
+            if isOn {
+                self.dateOnSaleStart = self.dateOnSaleStart != nil ? self.dateOnSaleStart : self.product.dateOnSaleStart ?? self.defaultStartDate
```

Comment:
super nit: I think this line (and similar for the line below) could also be
```
self.dateOnSaleStart = self.dateOnSaleStart ?? self.product.dateOnSaleStart ?? self.defaultStartDate
```
just to shorten a little 😄